### PR TITLE
feat: add agent-network page and core functionality

### PR DESCRIPTION
**Pull request by anette1983**
## Describe your changes

- Created the Agents Network Page with core functionality, connected it to database
- Implemented the chat notification section, Agent list component, search functionality, agents sorting
- Added draft chat and messages pages to test the link functionality

Знімок екрана 2023-12-24 192706
image
image
image
image

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?selectedIssue=AG-211)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/app/[locale]/agents/page.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+'use client';
+
+import { SnackbarProvider } from 'components/snackbar';
+import React from 'react';
+import AgentNetworkView from 'sections/agent-network-view';
```
Comment (AKutusenko) [inline src/app/[locale]/agents/page.tsx:5]: порядок импортов

File: src/app/[locale]/messages/page.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { randomValues } from 'constants/randomValues';
+
+function Messages() {
+  return <div>{randomValues.MESSAGES_PAGE}</div>;
```
Comment (AKutusenko) [inline src/app/[locale]/messages/page.tsx:5]: тип возвращаемого значения

**Review by AKutusenko — APPROVED** [review]

File: src/app/[locale]/agents/page.tsx (hunk 1 above)
Comment (anette1983) [reply src/app/[locale]/agents/page.tsx:5]: fixed

File: src/sections/chat/chat-msg-badge.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { useMemo } from 'react';
+import { useTranslations } from 'next-intl';
+import IconButton from '@mui/material/IconButton';
+import Badge from '@mui/material/Badge';
+import MailIcon from '@mui/icons-material/Mail';
+
+type MsgBadgeProps = {
+  tempValue: number;
```
Comment (lyudad) [inline src/sections/chat/chat-msg-badge.tsx]: tempValue это очень плохое название переменной, у тебя эта функция привязана к чатам, значит это не любая переменная а какая то специфическая. улучшить именование

Comment (anette1983) [reply src/sections/chat/chat-msg-badge.tsx]: fixed

**Pull request merged**